QualifiedTeachersApi/src/QualifiedTeachersApi/DataStore/Crm/DataverseAdapter.SetIttResultForTeacher.cs

```diff
@@ -329,8 +363,8 @@ public async Task<SetIttResultForTeacherLookupResult> LookupData()
             // in which case the status should be UnderAssessment, WithDrawn or Deferred.
 
             var inTrainingForProvider = ittRecords
-                .Where(r => (r.dfeta_ProgrammeType != dfeta_ITTProgrammeType.AssessmentOnlyRoute && (r.dfeta_Result == dfeta_ITTResult.InTraining || r.dfeta_Result == dfeta_ITTResult.Withdrawn || r.dfeta_Result == dfeta_ITTResult.Deferred) ||
-                            (r.dfeta_ProgrammeType == dfeta_ITTProgrammeType.AssessmentOnlyRoute && (r.dfeta_Result == dfeta_ITTResult.UnderAssessment || r.dfeta_Result == dfeta_ITTResult.Withdrawn || r.dfeta_Result == dfeta_ITTResult.Deferred))))
+                .Where(r => (r.dfeta_ProgrammeType != dfeta_ITTProgrammeType.AssessmentOnlyRoute && (r.dfeta_Result == dfeta_ITTResult.InTraining || r.dfeta_Result == dfeta_ITTResult.Withdrawn || r.dfeta_Result == dfeta_ITTResult.Deferred || r.dfeta_Result == dfeta_ITTResult.Pass) ||
```

Comment:
This is becoming so complex I wonder whether we should move this logic into a loop to make it easier to follow